### ISSUE TITLE
Compiler error for tutorial projects, Mac OS 10.13.

### ISSUE THREAD

**Issue by Forrest49** [orx/orx#30]
Hello,

I downloaded and installed orx per the Readme file, using setup.sh. I wanted to explore the tutorial projects in XCode, but they can't be built due to a linker error. It appears there may be some libraries Xcode is expecting that are absent? I looked through the forums but I was not able to find a solution.

Any help would definitely be appreciated!

**Comment by fgmcoelho** [comment]
Hey Forest, forums are not very active right now, I would encourage you to join our [gitter](https://gitter.im/orx/orx) channel, as most problems are solved there.

**Comment by iarwain** [comment]
Hi @Forrest49,

In the future you might have more luck asking your questions on [our forum](http://forum.orx-project.org) or directly on [gitter](https://gitter.im/orx/orx), as most of us might miss questions asked on github (but apparently I wrote too soon, I see that @fgmcoelho was faster than me!).

In any case, when you say you downloaded orx, did you use the source repository or one of the pre-built binary package?

**Comment by iarwain** [comment]
Yep, all good for the setup. Then I guess something changed in XCode since last time I was able to use it. 
As Apple has declared my computer to be too old to get recent OS/XCode updates, I can't check the issue on my side, unfortunately.
[Travis-ci doesn't report any building issue, but it's not using XCode directly, it's compiling using makefiles.](https://travis-ci.org/orx/orx/jobs/361979337)

We do have a [buildbot setup](http://buildbot.orx-project.org:8010) as well, but I'll have to check if any of our currently available slaves have a recent version of XCode (I know mine won't, but maybe another one will). I can't check that before tonight though. I'll keep you posted.

**Comment by iarwain** [comment]
I guess we'll have to address this warning at some point though:
`clang: warning: libstdc++ is deprecated; move to libc++ with a minimum deployment target of OS X 10.9 [-Wdeprecated]`

But right now we still support OS X back to 10.6, we'll have to change that first.

**Comment by Forrest49** [comment]
Thanks so much for the response! I'll continue rooting around and seeing if there's something I can do.

**Comment by iarwain** [comment]
My pleasure. I'd be happy to help more when I'm home tonight. If you can be on gitter in about 9-10 hours, I'm sure we'll find a way to sort it out! 😄
Otherwise, please let us know if you find the issue!

**Comment by Forrest49** [comment]
Will do! Last thing, just going to post the error message XCode is giving me, for posterity.
<img width="631" alt="errormessage" src="https://user-images.githubusercontent.com/8306931/38507651-14f9753e-3beb-11e8-9569-3d777b02666b.png">

**Comment by iarwain** [comment]
So apparently it doesn't find orx itself. Was there any errors in the orx library compilation?

**Comment by Forrest49** [comment]
Sorry, I didn't realize that I needed to compile the orx and orxLIB projects first. I just did that, they compiled in XCode successfully, so I went back to the Object tutorial, and now get a different error, shown below. 
<img width="766" alt="errormsg2" src="https://user-images.githubusercontent.com/8306931/38509081-ea90b420-3bee-11e8-8e89-6244345e499d.png">

So that's got something to do with a shell command not found? Might be making a bit of progress.

**Comment by iarwain** [comment]
Well, at least we're moving forward! I have no clue what that script could be. It looks like something that XCode would generate automatically. Have you tried to do a *rebuild* on the tutorials after you've built orx itself, not just a regular build?

**Comment by Arc676** [comment]
Can reproduce with the latest Xcode and macOS 10.13.4. The custom shell script isn't generated by Xcode, it's a post-build shell script included in the project file.
```
                9607AE3710C85E8F00CD1376 /* Postbuild */ = {
                        isa = PBXShellScriptBuildPhase;
                        buildActionMask = 2147483647;
                        files = (
                        );
                        inputPaths = (
                        );
                        name = Postbuild;
                        outputPaths = (
                        );
                        runOnlyForDeploymentPostprocessing = 0;
                        shellPath = /bin/sh;
                        shellScript = "$(shell [ -f ../../../../code/lib/dynamic/liborx.dylib ] && cp -f ../../../../code/lib/dynamic/liborx*.dylib ../../../bin)";
                };
```
Simply removing `shell` results in successful compilation, but the script copies both the release and debug libraries and for some reason it gives a warning about how GLFW things are defined in both libraries, even if I link the binary just to the release one.
```
Class GLFWWindowDelegate is implemented in both [...]/Orx/Source/tutorial/bin/liborx.dylib (0x100925c80) and [...]/Orx/Source/tutorial/bin/liborxd.dylib (0x1005999a0). One of the two will be used. Which one is undefined.
```

**Comment by iarwain** [comment]
That's what I meant: XCode is asked to perform a post-build step and it's XCode that generates the shell script on the fly that should contain the commands defined inside the project file. Why it's not generating the shell script, I can't say.
We're using premake4 to generate the XCode project, maybe the format has changed with the latest XCode? I'll look into it.
I also checked [the builds made by our two build machines](http://buildbot.orx-project.org:8010/builders/mac), and apparently they use XCode 7.1.1 and XCode 7.2 without any issue, so it's probably a format change that has happened after those.

In which step do you get the last message regarding `GLFWWindowDelegate`? This is very weird given that the release and debug version will never be used at the same time and I don't see what is special about `GLFWWindowDelegate`, if it were to warn about that issue, why not warn about all the other symbols as well? I'll try to google it to see what I can find.

**Comment by Arc676** [comment]
What is the `shell` command supposed to do? Removing it and just running
```
shellScript = "$([ -f ../../../../code/lib/dynamic/liborx.dylib ] && cp -f ../../../../code/lib/dynamic/liborx*.dylib ../../../bin)";
```
resulted in successful compilation.

There are multiple messages regarding GLFW being defined twice. I only included one, since they're all the same. I'm not sure why the binary would try to access both at the same time.

**Comment by iarwain** [comment]
I do not remember. It's been in our setup since 2012. The only references I found online tonight say it runs a command inside a shell, but I really can't say what makes it special nor why it was added in the first place.
However after testing it locally on linux/gmake, I found out the copy wouldn't happen without `shell`.
Are you sure the copy happened correctly without it in your setup?

**Comment by iarwain** [comment]
Removing the whole `$(shell` + matching `)` did seem to work though with linux/gmake, but I do not know if it'll work with all the supported IDEs as well.

**Comment by Arc676** [comment]
Yes; I tried deleting the `dylib` files from the `bin` directory and rebuilding and the copy was successful. Just the word `shell` though, not the surrounding `$()`.

I removed the binary linking option (since it wasn't there originally) and the GLFW warnings didn't appear (no idea why).

**Comment by Forrest49** [comment]
Sorry for the delay. I've done a clean build on the tutorials.

**Comment by iarwain** [comment]
Thanks, @Arc676. If you could try doing the change in the `premake4.lua` file directly and submit a pull request, that'd be great as I can't test locally.

**Comment by iarwain** [comment]
Is the latest working for you now, @Forrest49?
